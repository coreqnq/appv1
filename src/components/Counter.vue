<template>
<h2>Section Counter</h2>
  <h1>{{ getTitle }} : {{squareCounter }}</h1>
  <p>{{counter}}<sub>2</sub>   = {{squareCounter}}</p>
  <p data-testid="modelID">{{counter}}</p>

  <div class="container">
    <button @click="increase">+1</button>
    <button @click="decrease">-1*</button>
  </div>
</template>
<script>
export default {
  props:
{
  titleBook : String,
      start : { type: Number,
                // require: true,
                default: 100
      }
}
  ,
  name: "Counter",
  data () {
    return {
      counter: this.start
    }
  },
  methods :{
    getSquare() {
      return this.counter * this.counter
    },

    increase() {
      this.counter ++;
    },
    decrease() {
      this.counter --;
    }
  },
  computed : {
    squareCounter () {
      console.log('time')
      return this.counter * this.counter
    },
    getTitle(){
      return this.titleBook || 'Default'
    }
  }

}
</script>

<style scoped>

  button{
    background-color: #64BB87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
  }

  button:hover{
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;

  }

</style>