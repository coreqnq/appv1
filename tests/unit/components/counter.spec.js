import { shallowMount } from '@vue/test-utils'
import Counter  from "@/components/Counter";

describe('Counter Componet' , () => {

    let wrapper = shallowMount(Counter)

    beforeEach(() => {
        wrapper = shallowMount(Counter)
    })

    test('debe de hacer match con el snapshot', () => {
        // const wrapper = shallowMount(Counter)

        expect( wrapper.find('h2').exists() ).toBeTruthy()

        const h2Value = wrapper.find('h2').text()

        expect(h2Value).toBe('Section Counter')
    })

    test('p map 2 test', () => {
        // const wrapper = shallowMount(Counter)
        const pTags = wrapper.find('[data-testid="modelID"]').text()
        expect(pTags).toBe('100')

    })

    test('debe de incremetar el 1 el valor del contador', async() => {
        // const wrapper = shallowMount(Counter)

        const [increaseBtn , decreaseBtn] = wrapper.findAll('button')

        await increaseBtn.trigger('click')
        await increaseBtn.trigger('click')
        await increaseBtn.trigger('click')

        await decreaseBtn.trigger('click')
        await decreaseBtn.trigger('click')
        // await increaseBtn.trigger('click')


        // let bTags = wrapper.find('[data-testid="modelAcctionID"]').text()
        // const value = wrapper.find('[data-testid="modelID"]').text()

        const value = wrapper.find('[data-testid="modelID"]').text()

        // expect(value).toBe('101')
        expect(value).toBe('101')
    })

    test('debe de ser establecer el valor por defecto', () => {
        let {start} = wrapper.props();
        const value = wrapper.find('[data-testid="modelID"]').text()
        expect(Number(value)).toBe(start);
    })

    test('debe de ser establecer el valor por defecto', () => {
        let  wrapper =   shallowMount(Counter);

    })



})