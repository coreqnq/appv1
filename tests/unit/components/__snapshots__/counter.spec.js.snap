// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Componet debe de hacer match con el snapshot 1`] = `
<h2>Section Counter</h2>
<h1>Default : 100</h1>
<div><button>+1</button><button>-1</button></div>
`;
